added fill value to Cutout constructor

bfialkoff opened the pull request:
I had the need to use the cutout augmentation, but I wanted to be able to set the color. As it was, the cutout was always black. I thought it would be a useful feature to be able to set the fill_value.

kirillbobyrev commented:
Hi! I have recently proposed `Cutout` improvements in #263 and this looks like a subcase for the third improvement (patch modification). I've implemented `CoarseDropout` which deprecates `Cutout` in #264 and I think this patch should at least be applied to `CoarseDropout`. If you could find a way to generalize it and build a high-level interface for patch modifications that could be even better, though I don't have much experience with the project so I don't know what would be the best way to do that.

ternaus commented:
Does not pass the tests.

bfialkoff commented:
Yes, I know I made some changes that caused the fail and just haven't
gotten around to correcting it yet.

On Thu, Jun 6, 2019, 10:27 PM Vladimir Iglovikov <notifications@github.com>
wrote:

> Does not pass the tests.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/albu/albumentations/pull/267?email_source=notifications&email_token=AJO6FM2PCOSVUPM3GJ3VPJ3PZFQJHA5CNFSM4HP2PFR2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODXD5ASY#issuecomment-499634251>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AJO6FMYMUAT5GM7DGEQ3IJLPZFQJHANCNFSM4HP2PFRQ>
> .
>

John-Yao commented:
hello, I had tried to set the value of Cutout's fill_value . It does work well. All the cut hole  is fill with black.

bfialkoff commented:
That's strange. Please sharead your code and I'll try and diagnose the
problem.

On Tue, Jul 2, 2019, 5:38 AM John-Yao <notifications@github.com> wrote:

> hello, I had tried to set the value of Cutout's fill_value . It does work
> well. All the cut hole is fill with black.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/albu/albumentations/pull/267?email_source=notifications&email_token=AJO6FM45JIDYNTZ6VKQJGL3P5K5RRA5CNFSM4HP2PFR2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODY74CUQ#issuecomment-507494738>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AJO6FM7O7R4WERVBAKNL243P5K5RRANCNFSM4HP2PFRQ>
> .
>

John-Yao commented:
import sys
import cv2
import albumentations as A
import matplotlib.pyplot as plt
from mmcv.runner import obj_from_dict
print(A.__version__)
class Augtor(object):
    
    def __init__(self,
                 albumentations = None):
        self.abmtt_transforms = []
        
        if albumentations is not None:
            for _transform in albumentations:                
                self.abmtt_transforms.append(obj_from_dict(_transform, parent=A))
                
    def __call__(self, img):
        img = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)
        for transform in self.abmtt_transforms: #rgb
            img = transform(image=img)['image']
        img = cv2.cvtColor(img, cv2.COLOR_RGB2BGR)
        return img
sampler = Augtor(albumentations=[
               dict(
                    max_h_size=100,
                    max_w_size=100,
                    p=1.0,
                    num_holes = 2,
                    fill_value=[255,0,0],#rgb
                    type='Cutout'
                )
])
I = cv2.imread(img_fname)
sample_img = sampler(I[:1000,:1000])
plt.imshow(cv2.cvtColor(sample_img,cv2.COLOR_BGR2RGB).astype(np.int))
plt.show()

Thanks for your reply! the 'img_fname' have to be set by yourself.
The following image is my result.

![image](https://user-images.githubusercontent.com/22048231/60482026-81fde780-9cc2-11e9-887f-dc7050700b10.png)

albu commented on albumentations/core/transforms_interface.py:
```diff
@@ -95,7 +95,11 @@ def targets(self):
             params['interpolation'] = self.interpolation
-        params.update({'cols': kwargs['image'].shape[1], 'rows': kwargs['image'].shape[0]})
+        if hasattr(self, 'fill_value'):
+            params['interpolation'] = self.fill_value
```
it's strange to set interpolation as fill value

bfialkoff replied:
Not strange, just plain incorrect! That's what I get for copy-pasting. I
fixed it and updated the PR.

bfialkoff commented:
Not strange, just plain incorrect! That's what I get for copy-pasting. I
fixed it and updated the PR.

On Tue, Jul 2, 2019, 9:34 AM Alexander Buslaev <notifications@github.com>
wrote:

> *@albu* commented on this pull request.
> ------------------------------
>
> In albumentations/core/transforms_interface.py
> <https://github.com/albu/albumentations/pull/267#discussion_r299321148>:
>
> > @@ -95,7 +95,11 @@ def targets(self):
>      def update_params(self, params, **kwargs):
>          if hasattr(self, 'interpolation'):
>              params['interpolation'] = self.interpolation
> -        params.update({'cols': kwargs['image'].shape[1], 'rows': kwargs['image'].shape[0]})
> +        if hasattr(self, 'fill_value'):
> +            params['interpolation'] = self.fill_value
>
> it's strange to set interpolation as fill value
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/albu/albumentations/pull/267?email_source=notifications&email_token=AJO6FM4IGM37Z4EW6VYB64DP5LZHDA5CNFSM4HP2PFR2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOB5GL6RI#pullrequestreview-256687941>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AJO6FM6JFM66JAHWXVNNOTLP5LZHDANCNFSM4HP2PFRQ>
> .
>
